enzyme/index.d.ts

```diff
@@ -541,14 +541,14 @@ export interface MountRendererProps {
  * @param node
  * @param [options]
  */
-export function shallow<P, S>(node: ReactElement<P>, options?: ShallowRendererProps): ShallowWrapper<P, S>;
+export function shallow<P, S>(node: ReactElement<P>, options?: ShallowRendererProps): ShallowWrapper<P, any>;
 
```

Comment:
No. It could be used with following:
```tsx
let wrapper: ShallowWrapper<Props, State> = shallow(<SomeComponent />);
wrapper.state().prop;
```

The problem that ```shallow()``` can't infer the correct type for State from ```<MyComponent />``` so it's better to treat it as any.